pkg/awsutils/awsutils.go

```diff
@@ -698,6 +710,38 @@ func (cache *EC2InstanceMetadataCache) tagENI(eniID string) {
 	})
 }
 
+//parseAdditionalEniTagsMap will create a map for additional tags
+func parseAdditionalEniTagsMap(additionalEniTags string) (map[string]string, error) {
+	var additionalEniTagsMap map[string]string
+
+	// If duplicate keys exist, the value of the key will be the value of latter key.
+	err := json.Unmarshal([]byte(additionalEniTags), &additionalEniTagsMap)
+	if err != nil {
+		log.Errorf("Invalid format for ADDITIONAL_ENI_TAGS. Expected a json hash: %v", err)
+	}
+	return additionalEniTagsMap, err
+}
+
+// MapToTags converts a map to a slice of tags.
+func mapToTags(tagsMap map[string]string, tags []*ec2.Tag) []*ec2.Tag {
+	if tagsMap == nil {
+		return tags
+	}
+
+	for key, value := range tagsMap {
+		keyPrefix := reservedTagKeyPrefix
+		if strings.Contains(key, keyPrefix) {
```

Comment:
This is kind of misleading (same goes for the error message below) as this is not a prefix check.